src/utils.rs

```diff
@@ -30,6 +31,16 @@ pub enum Error {
     },
 }
 
+/// Helper function to map pipeline errors to library errors
+pub fn map_pipeline_error(err: pipeline::Error) -> crate::Error {
+    match err {
+        pipeline::Error::ScanOperation(scan_err) => crate::Error::ScanOperation(scan_err),
+        pipeline::Error::MirrorOperation(mirror_err) => crate::Error::MirrorOperation(mirror_err),
+        pipeline::Error::Io(io_err) => crate::Error::Io(io_err),
+        other => crate::Error::Other(other.to_string()),
+    }
+}
```

Comment:
Fyi if you move this logic into a `impl From<pipeline::Error> for crate::Error` then all of the places like `.map_err(map_pipeline_error)` that it gets used should be replaceable with `?`.